core/embed/extmod/modtrezorcrypto/modtrezorcrypto-secp256k1_zkp.h

```diff
@@ -321,22 +411,284 @@ STATIC mp_obj_t mod_trezorcrypto_secp256k1_context_multiply(
+  if (num_of_inputs < 1 || num_of_inputs >= n_commitments) {
+    mp_raise_ValueError("Invalid number of inputs");
+  }
+
+  if (!secp256k1_pedersen_verify_tally(ctx, commitments, num_of_inputs,
+                                       commitments + num_of_inputs,
+                                       n_commitments - num_of_inputs)) {
+    mp_raise_ValueError("Pedersen commitments are not balanced");
+  }
+
+  for (int i = 0; i < n_commitments; ++i) {
+    m_del_obj(secp256k1_pedersen_commitment, (void *)(commitments[i]));
+  }
+  return mp_const_none;
+}
+
+STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(
+    mod_trezorcrypto_secp256k1_context_verify_balance_obj, 3, 3,
+    mod_trezorcrypto_secp256k1_context_verify_balance);
+
+/// def rangeproof_sign(self, config: RangeProofConfig, value: int,
+///                     commit: bytes, blind: bytes, nonce: bytes,
+///                     message: bytes, extra_commit: bytes,
+///                     gen: bytes) -> memoryview:
+///     '''
+///     Return a range proof for specified value (as a memoryview of the
+///     underlying rangeproof_buffer).
+///     '''
+STATIC mp_obj_t mod_trezorcrypto_secp256k1_context_rangeproof_sign(
+    size_t n_args, const mp_obj_t *args) {
+  mp_obj_secp256k1_context_t *o = MP_OBJ_TO_PTR(args[0]);
+  if (o->secp256k1_ctx == NULL || o->rangeproof_buffer == NULL) {
+    mp_raise_msg(&mp_type_RuntimeError, "not entered secp256k1_zkp.Context");
+  }
+  secp256k1_context *ctx = o->secp256k1_ctx;
+  uint8_t *rangeproof_buffer = o->rangeproof_buffer;
+
+  mp_obj_range_proof_config_t *config = MP_OBJ_TO_PTR(args[1]);
+
+  const uint64_t value = trezor_obj_get_uint64(args[2]);
+
+  secp256k1_pedersen_commitment commitment;
+  parse_commitment(ctx, &commitment, args[3]);
+
+  mp_buffer_info_t blind;
+  mp_get_buffer_raise(args[4], &blind, MP_BUFFER_READ);
+  if (blind.len != 32) {
+    mp_raise_ValueError("Invalid length of blinding factor");
+  }
+
+  mp_buffer_info_t nonce;
+  mp_get_buffer_raise(args[5], &nonce, MP_BUFFER_READ);
+  if (nonce.len != 32) {
+    mp_raise_ValueError("Invalid length of nonce");
+  }
+
+  mp_buffer_info_t message;
+  mp_get_buffer_raise(args[6], &message, MP_BUFFER_READ);
+
+  mp_buffer_info_t extra_commit;
+  mp_get_buffer_raise(args[7], &extra_commit, MP_BUFFER_READ);
+
+  secp256k1_generator generator;
+  parse_generator(ctx, &generator, args[8]);
+
+  memzero(rangeproof_buffer, RANGEPROOF_SIGN_BUFFER_SIZE);
+  size_t rangeproof_len = RANGEPROOF_SIGN_BUFFER_SIZE;
+
+  if (!secp256k1_rangeproof_sign(
+          ctx, rangeproof_buffer, &rangeproof_len, config->min_value,
+          &commitment, blind.buf, nonce.buf, config->exponent, config->bits,
+          value, message.buf, message.len, extra_commit.buf, extra_commit.len,
+          &generator)) {
+    mp_raise_ValueError("Rangeproof sign failed");
+  }
+  if (rangeproof_len > RANGEPROOF_SIGN_BUFFER_SIZE) {
+    mp_raise_ValueError("Too large rangeproof");
```

Comment:
If this happens then we have serious bug in secp256k1-zkp. I think you should just remove this check. Or die here instead of raising a simple ValueError (in that case, we have probably overwritten memory we shouldn't access)

Reply:
You're right - removed the check.